k2/csrc/array_ops_benchmark.cu

```diff
@@ -0,0 +1,98 @@
+/**
+ * @brief Benchmarks for k2 APIs.
+ *
+ * @copyright
+ * Copyright (c)  2020  Xiaomi Corporation (authors: Fangjun Kuang)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#include "k2/csrc/array_ops.h"
+#include "k2/csrc/log.h"
+#include "k2/csrc/timer.h"
+
+namespace k2 {
+
+// ------- from google/benchmark  ---- begin ----
```

Comment:
We should probably put this generic stuff in something like csrc/benchmark/benchmark.h, and I think it would be good to put the benchmarks in csrc/benchmark/array_ops_benchmark.cu.  To separate it from the other stuff.

And I think there should be a function to print out relevant info about the CPU and GPU, and the date and time, and maybe certain key compiler options (at least debug/release mode)  at the beginning of running the program- so we can keep track of this stuff over time.  What do you think?
I know we'll mostly be comparing this on the same machine before and after a change, but we may also want to make other kinds of comparisons and do various kinds of data analytics later on, e.g. to find problematic hardware.

Reply:
Yes, I agree.

Reply:
> And I think there should be a function to print out relevant info about the CPU and GPU, and the date and time,

I've printed some info about GPU. Not sure what kind of GPU information is needed.